mlrun/api/crud/secrets.py

```diff
@@ -95,7 +104,38 @@ def store_secrets(
                 f"Provider requested is not supported. provider = {secrets.provider}"
             )
 
-    def delete_secrets(
+    def store_auth_secret(
+        self,
+        secret: mlrun.api.schemas.AuthSecretData,
+    ) -> str:
+        if secret.provider != mlrun.api.schemas.SecretProviderName.kubernetes:
+            raise mlrun.errors.MLRunInvalidArgumentError(
+                f"Storing auth secret is not implemented for provider {secret.provider}"
+            )
+        if not mlrun.api.utils.singletons.k8s.get_k8s():
+            raise mlrun.errors.MLRunInternalServerError(
+                "K8s provider cannot be initialized"
+            )
+        return mlrun.api.utils.singletons.k8s.get_k8s().store_auth_secret(
```

Comment:
I think the auth logic (what secret name to use, how the `AuthSecretData` class is constructed etc.) should be in this class. The `k8s_utils` class IMO should only provide low-level k8s secret operations. For example, I would place the `get_auth_secret_name()` function in this class etc.
Was there a reason to place the logic in the `k8s_utils` class?

Reply:
The reason is that it's a provider specific logic, if we'll support another provider for the auth secret, ideally the `store_auth_secret` interface will stay the same